code/game/gamemodes/meteor/meteors.dm

```diff
@@ -39,24 +39,22 @@
 	var/obj/effect/meteor/M = new Me(pickedstart)
 	M.dest = pickedgoal
 	M.z_original = 1
-	spawn(0)
-		walk_towards(M, M.dest, 1)
-	return
+	addtimer(GLOBAL_PROC, "walk_towards", 0, FALSE, M, M.dest, 1)
```

Comment:
Why are you using the timer SS like this and who let you

spawn(0)'s are made for hacky pseudo parallelism, not for scheduling a task